src/toil/batchSystems/kubernetes.py

```diff
@@ -211,7 +232,21 @@ def _api(self, kind, max_age_seconds = 5 * 60):
                 return self._apis[kind]
             except KeyError: 
                 raise RuntimeError("Unknown Kubernetes API type: {}".format(kind))
+    
+    def _try_kubernetes(self, method, *args, **kwargs):
+        """
+        Kubernetes API can end abruptly when it could wait and retry until successful or terminate immediatley if it encounters 
+        an unretryable error such as gateway timeout or max number of retries reached. 
+        For example, in a traditional case where we call self._api('batch').create_namespaced_job(self.namespace, job)
+        without this wrapper, Kubernetes can behave irregularly and inconsistently given a large job.
         
+        This new function, given a large volume of incoming jobs, we can give Kubernetes more time in between tries 
+        to check if the API is executable or not by calling self._try_kubernetes(self._api('batch').list_namespaced_job, self.namespace, **kwargs).      
```

Comment:
This docstring sort of communicates the point of the function, but these are not really good sentences.

It sounds like a gateway timeout is unretryable, for example, but this function does indeed retry when there is a gateway timeout.

Also, the last sentence has two subjects ("This new function" and "we").